dapp/src/utils/orangekit/ledger-live/bitcoin-provider.ts

```diff
@@ -0,0 +1,322 @@
+  tryConnectToAddress: string | undefined
+}
+
+/**
+ * Ledger Live Wallet API Bitcoin Provider.
+ */
+export default class AcreLedgerLiveBitcoinProvider
+  implements OrangeKitBitcoinWalletProvider, AcreBitcoinProvider
+{
+  readonly #walletApiClient: WalletAPIClient<
+    (client: WalletAPIClient) => { acre: AcreModule }
+  >
+
+  readonly #windowMessageTransport: WindowMessageTransport
+
+  readonly #network: BitcoinNetwork
+
+  readonly #options: LedgerLiveWalletApiBitcoinProviderOptions
+
+  readonly #hwAppId = "Acre"
+
+  readonly #derivationPath = "0/0"
+
+  #hasConnectFunctionBeenCalled: boolean = false
+
+  #account: Account | undefined
+
+  static init(
+    network: BitcoinNetwork,
+    options: LedgerLiveWalletApiBitcoinProviderOptions = {
+      tryConnectToAddress: undefined,
+    },
+  ) {
+    const windowMessageTransport = new WindowMessageTransport()
+    windowMessageTransport.connect()
+
+    const walletApiClient = new WalletAPIClient(
+      windowMessageTransport,
+      defaultLogger,
+      (client: WalletAPIClient) => ({ acre: new AcreModule(client) }),
+    )
+
+    return new AcreLedgerLiveBitcoinProvider(
+      network,
+      windowMessageTransport,
+      walletApiClient,
+      options,
+    )
+  }
+
+  constructor(
+    network: BitcoinNetwork,
+    windowMessageTransport: WindowMessageTransport,
+    walletApiClient: WalletAPIClient,
+    options: LedgerLiveWalletApiBitcoinProviderOptions = {
+      tryConnectToAddress: undefined,
+    },
+  ) {
+    this.#network = network
+    this.#windowMessageTransport = windowMessageTransport
+    this.#walletApiClient = walletApiClient
+    this.#options = options
+  }
+
+  /**
+   * Connects with the Ledger Live Wallet API.
+   * @returns Connected account address.
+   */
+  async connect(): Promise<BitcoinAddress> {
+    this.#windowMessageTransport.connect()
+
+    const currencyIds = [
+      this.#network === BitcoinNetwork.Mainnet ? "bitcoin" : "bitcoin_testnet",
+    ]
+
+    if (!this.#hasConnectFunctionBeenCalled) {
+      const accounts: Account[] = await this.#walletApiClient.account.list({
+        currencyIds,
+      })
+
+      if (this.#options.tryConnectToAddress) {
+        for (let i = 0; i < accounts.length; i += 1) {
+          const acc = accounts[i]
+          // eslint-disable-next-line no-await-in-loop
+          const address = await this.#getAddress(acc.id)
+
+          if (address === this.#options.tryConnectToAddress) {
+            this.#account = acc
+            break
+          }
+        }
+      } else if (accounts.length === 1) {
+        ;[this.#account] = accounts
+      }
+    }
+
+    if (!this.#account || this.#hasConnectFunctionBeenCalled) {
+      this.#account = await tryRequest<Account>()(async () =>
+        this.#walletApiClient.account.request({
+          currencyIds,
+        }),
+      )
+    }
+
+    this.#hasConnectFunctionBeenCalled = true
+    return this.#account.address
```

Comment:
Shouldn't we call `#getAddress` here to return the address at 0 index?

Reply:
5de537c